TROUBLESHOOTING.md

```diff
@@ -202,6 +202,19 @@ ruby-lsp
 Is there any extra information given from booting the server manually? Or does it only fail when booting through the
 extension?
 
+## Indexing ##
+
+When Ruby LSP starts, it attempts to index your code as well as your dependencies as described in [Configuring code indexing](README.md#configuring-code-indexing).
+
+In rare cases, Ruby LSP will encounter an error which prevents parsing from completing, meaning Ruby LSP cannot start.
```

Comment:
We rescue all indexing errors, so that the language server doesn't crash. The only consequence of an indexing failure is incomplete indexing data.
```suggestion
In rare cases, Ruby LSP will encounter an error which prevents indexing from completing, which will result in incomplete information in the editor.
```